transaction/core/src/input_rules.rs

```diff
@@ -85,6 +85,11 @@ impl InputRules {
         result
     }
 
+    /// Compute MCIP-52 canonical digest of input rules
+    pub fn canonical_digest(&self) -> [u8; 32] {
```

Comment:
:question: Is there a cryptographic domain usage for *canonical*?
Seeing it's usage implies that there is another *non_canonical* digest in use somewhere?

Reply:
Yea, its usually important for bytes / digests to be canonical when they are being signed or verified in some way.

One of the things mc-crypto-digestible achieves is that it produces canonical digests of objects regardless of any issues around serialization. For comparison, protobuf serializations are not canonical -- there are many valid serializations of an object and protobuf encoders can change these bytes without it being considered a breaking change, as long as its forwards and backwards compatible. So digesting protobuf bytes is not a canonical digest. This one is, and that's what the word canonical is meant to signal. Changing the bytes of this digest at all for a given object is considered an (unacceptable) breaking change.